### DISCUSSION TITLE
Can't return result from session.exec directly (have to assign to an intermediate variable)?

### DISCUSSION THREAD

**Discussion by EarthStoodStill** [fastapi/sqlmodel#984] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

I now know from the examples (after looking more) that you always assign the result from session.exec calls to a var of some Pydantic model type before returning anything from a function, but I'm still curious why the former doesn't work below (but the latter does). Can I get some info (maybe which library leads to this behavior, and where I can look to learn more about why this is needed, or what I can search for to read some more, since they appeared functionally equivalent to me).

```python
# DOESN'T WORK -- causes an internal 500 error/session.exec(...) returns None
@app.get(f'/organizations/', response_model=List[models.Organization])
def read_organizations() -> List[models.Organization]:
    with Session(engine) as session:
        return session.exec(
            select(models.Organization)
        ).all()
```
vs.

```py
# WORKS (returns list of organizations)
@app.get(f'/organizations/', response_model=List[models.Organization])
def read_organizations() -> List[models.Organization]:
    with Session(engine) as session:
        organizations: List[models.Organization] = session.exec(
            select(models.Organization)
        ).all()

        return organizations
```

### Description

Just trying to get back all "organizations" instead of a single one. Something about my code structure in the former code is maybe returning from session.exec(...) before it has data, or is breaking it (but this is synchronous, so I wouldn't think that's the case)? I noticed you always assign to an intermediate variable before returning anything, so hopefully you know why/can point me to something I can read/Google about.

### Operating System

Linux

### SQLModel Version

0.0.18

### Python Version

3.12

### Additional Context

Thank you! Sorry if this is just a dumb variable scope/Python code structure issue -- I'm a bit rusty atm.

**Comment by EarthStoodStill** [comment]
p.s., I suppose the Pydantic types are being constructed from the result data -- maybe that's the answer. I wasn't expecting that to be necessary since I passed the type to the embedded select(...) (so maybe they don't come back as my custom Pyndatic types from session.exec(...)?)

**Comment by YuriiMotov** [comment]
There shouldn't be any difference. Could you provide MRE?

<details>

```py
from typing import List

from fastapi import FastAPI
from sqlalchemy import StaticPool
from sqlmodel import Field, Session, SQLModel, create_engine, select


class Organization(SQLModel, table=True):
    id: int = Field(default=None, primary_key=True)
    name: str


# Create an in-memory SQLite database
connect_args = {"check_same_thread": False}
engine = create_engine(
    "sqlite:///:memory:", connect_args=connect_args, poolclass=StaticPool
)
SQLModel.metadata.create_all(engine)

with Session(engine) as session:
    session.add(Organization(name="Example Org"))
    session.add(Organization(name="Example Org 2"))
    session.add(Organization(name="Example Org 3"))
    session.commit()

app = FastAPI()


@app.get("/organizations/")
def read_organizations() -> List[Organization]:
    with Session(engine) as session:
        return session.exec(select(Organization)).all()
```

</details>
